src/parser/uniterm.rs

```diff
@@ -171,6 +174,40 @@ impl UniRecord {
         // `pos_ellipsis` should be set iff `attrs.open` is true.
         debug_assert!((self.pos_ellipsis == TermPos::None) != self.attrs.open);
 
+        fn term_to_record_rows(
```

Comment:
nitpick: I would put this function at the beginning of `into_type_strict`. It's slightly odd to have some code (the debug assert), then a local function definition, then code again.